### PR TITLE
Empty .web/pages directory on recompile

### PR DESCRIPTION
**Pull request by andy-verstraeten**
Empty the .web/pages directory on recompile so that old pages that don't exist anymore in the python code get removed.

### CODE REVIEW

File: pynecone/compiler/compiler.py [hunk 1]
```diff
@@ -208,3 +208,9 @@ def compile_components(components: Set[CustomComponent]):
     # Compile the components.
     code = _compile_components(components)
     return output_path, code
+
+
+def purge_web_pages_dir():
+    """Empty out .web directory.
+    """
+    utils.empty_dir(constants.WEB_PAGES_DIR)
```
Comment (picklelo) [inline pynecone/compiler/compiler.py:216]: There's two special files in the pages directory - `._app.js` and `404.js` that are part of the template and not recompiled. I think we need to exclude those two files when purging the dir.

**Review by picklelo — CHANGES_REQUESTED** [review]
Thanks for digging into this!

**Comment by andy-verstraeten** [comment]
Both files were added to an ignore list.

File: pynecone/compiler/compiler.py [hunk 2]
```diff
@@ -208,3 +208,10 @@ def compile_components(components: Set[CustomComponent]):
     # Compile the components.
     code = _compile_components(components)
     return output_path, code
+
+
+def purge_web_pages_dir():
+    """Empty out .web directory.
+    """
+    template_files = ["_app.js","404.js"]
```
Comment (picklelo) [inline pynecone/compiler/compiler.py:216]: I think you forgot to pass this in as an argument

File: pynecone/compiler/utils.py [hunk 3]
```diff
@@ -304,3 +304,17 @@ def write_page(path: str, code: str):
...
+
+def empty_dir(path, ignored=[]):
+    """Remove all files and folders in a directory except for the ignored file- or foldernames.
+
+    Args:
+        path (str): The path to the directory that will be emptied
+        excluded (list, optional): List of filenames or foldernames that will not be deleted. Defaults to [].
+    """
+    directory_contents = os.listdir(path)
+    print(directory_contents)
```
Comment (picklelo) [inline pynecone/compiler/utils.py:316]: We can remove this print

**Comment by andy-verstraeten** [comment]
My bad.

**Review by picklelo — APPROVED** [review]
Awesome!

**Comment by picklelo** [comment]
Sorry, needs one more change to make the linter happy (I think it wants the docstring on a single line). Can you make sure the following checks pass:
```
poetry run pyright pynecone tests
poetry run pydocstyle pynecone tests
find pynecone tests -name "*.py" -not -path pynecone/pc.py | xargs poetry run darglint
poetry run black pynecone tests
```

**Comment by picklelo** [comment]
Thank you for the persistence 😄

**Pull request merged**